input/system/azure/system.go

```diff
@@ -0,0 +1,282 @@
+		server.SelfTest.HintCollectionAspect(state.CollectionAspectSystemStats, "Make sure the Monitoring Reader permission of the database is granted to the managed identity.")
+		return
+	}
+
+	// Server metrics
+	client, err := azquery.NewMetricsClient(credential, nil)
+	if err != nil {
+		server.SelfTest.MarkCollectionAspectError(state.CollectionAspectSystemStats, "error making a metrics client: %v", err)
+		logger.PrintError("Azure/System: Failed to make a metrics client: %v\n", err)
+		return
+	}
+
+	// Query metrics data with 1 min interval, for the last 1 min (should return 1 value)
+	metricNames := "cpu_percent,memory_percent,network_bytes_egress,network_bytes_ingress,storage_used"
+	if strings.ToLower(system.Info.Azure.ResourceType) == "flexibleservers" {
+		// metrics only available with Flexible Server
+		metricNames += ",txlogs_storage_used,read_iops,write_iops,disk_queue_depth,read_throughput,write_throughput"
+	}
+	option := &azquery.MetricsClientQueryResourceOptions{
+		MetricNames: to.Ptr(metricNames),
+		Aggregation: to.SliceOfPtrs(azquery.AggregationTypeAverage),
+		Interval:    to.Ptr("PT1M"),
+		Timespan:    to.Ptr(azquery.TimeInterval("PT1M")),
+	}
+
+	metricsRes, err := client.QueryResource(ctx, resourceID, option)
+	if err != nil {
+		server.SelfTest.MarkCollectionAspectError(state.CollectionAspectSystemStats, "error getting server metrics: %v", err)
+		logger.PrintError("Azure/System: Encountered error getting server metrics: %v\n", err)
+		server.SelfTest.HintCollectionAspect(state.CollectionAspectSystemStats, "Make sure the Monitoring Reader permission of the database is granted to the managed identity.")
+		return
+	}
+
+	system.CPUStats = make(state.CPUStatisticMap)
+	diffedNetworkStats := &state.DiffedNetworkStats{}
+	diffedDiskStats := &state.DiffedDiskStats{}
+	for _, metric := range metricsRes.Value {
+		// Should be only one data as 1 min time span with 1 min interval is selected, so getting first metric is good
+		metricValue := getFirstMetricValue(metric)
+		if metricValue.Average == nil {
+			continue
+		}
+		switch *metric.Name.Value {
+		case "cpu_percent":
+			system.CPUStats["all"] = state.CPUStatistic{
+				DiffedOnInput: true,
+				DiffedValues: &state.DiffedSystemCPUStats{
+					UserPercent: *metricValue.Average,
+				},
+			}
+		case "memory_percent":
+			// Currently, we can only retrieve memory percent
+			// Since total memory size is not listed in the server info,
+			// we are unable to pass this value to pganalyze at the moment
+		case "txlogs_storage_used":
+			system.XlogUsedBytes = uint64(*metricValue.Average)
+		case "network_bytes_egress":
+			// value is total of 1min
+			diffedNetworkStats.TransmitThroughputBytesPerSecond = uint64(*metricValue.Average / 60)
+		case "network_bytes_ingress":
+			// value is total of 1min
+			diffedNetworkStats.ReceiveThroughputBytesPerSecond = uint64(*metricValue.Average / 60)
+		case "read_iops":
+			diffedDiskStats.ReadOperationsPerSecond = *metricValue.Average
```

Comment:
Sentry reported a nil pointer dereference here https://sentry.io/issues/5910271246

Reply:
Yeah it's already fixed actually, but thanks for the ping!